faunadb-scala/src/main/scala/faunadb/values/Codec.scala

```diff
@@ -35,6 +42,30 @@ object Decoder {
       }
   }
 
+  implicit object CharDecoder extends Decoder[Char] {
```

Comment:
Also I didn't get what bug in the system you are talking about. The only think I can imagine is if the value stored in fauna overflow the representation of the destination type, like `LongV(1000000).to[Char]`, it will cause the `toChar` function to throw an runtime exception, but I think that is reasonable. Also all the type checking will be done a compile time, could you provide some example of what bug you are talking about?